maps/generic_map.go

```diff
@@ -56,6 +56,15 @@ func (m Map[K, V]) IsEmpty() bool {
 	return len(m) == 0
 }
 
+// Clone the current map
+func (m Map[K, V]) Clone() Map[K, V] {
+	clone := make(Map[K, V])
```

Comment:
what do you think about using `maps.Clone(...)` from `golang.org/x/exp/maps`? It preserves nil and perform precise allocation.

Reply:
i agree, should we keep the method and use `maps.Clone` internally or remove it altogether ??

Reply:
I think we can keep it and use it internally. Probably we can also implement it for sync lock map wrapped with a mutex so that it prevents race conditions